gandi_config/config.go

```diff
@@ -0,0 +1,7 @@
+package gandi_config
```

Comment:
Suggestion: Delete gandi_config/config.go and put `type Config` directly in gandi.go.

config.go includes only one struct, and it can only be used in gandi.go.  Why have it be in a separate file?  is there any case where one would want the config and not also be importing gandi.go?

The reason I suggest this is that I'm updating my code to use the new (WIP) module.  It seems like I'm importing gandi_config any time I use this module.

Reply:
I completely agree.

Reply:
I started from there, but that results in an import cycle if the constructors are also in that package..

Reply:
(because the constructors import `gandi_domain`, but `gandi_domain` needs `gandi` for the Config struct)

Reply:
I'd suggest instantiating the client in the `NewXXXClient` functions, and calling the `NewFromClient` functions only, and when instantiating the client, passing the config arguments as individual arguments, not a struct.

By the way, the only "real" config here is `SharingID`. `DryRun` is not used, and `Debug` may probably be replaced by something other than `fmt.Println` (I'm thinking about using a log library, but something else may exist) :)

Reply:
Refactoring Config{}: That's a really good idea. I wanted to see what that would look like, so I started coding a little. Because I'm procrastinating from doing my day-job, I ended up finishing the change and made a PR.  Please feel free to ignore or steal as much as you want from https://github.com/thommay/go-gandi-livedns/pull/2  (I apologize in advance if that's stepping on your toes.)

I agree that the only "real" config is SharingID.   I think Debug using fmt.Println is fine for now... it can be replaced later.  Since DryRun isn't used, I removed it from the struct.

Reply:
DryRun appears to be available on a call by call basis (ie, https://api.gandi.net/docs/domains/#post-v5-domain-domains supports it), so I think at this point it probably makes more sense to make that explicit and expose it on a function by function basis.